bridges/BrutBridge.php

```diff
@@ -0,0 +1,128 @@
+<?php
+class BrutBridge extends BridgeAbstract {
+	const NAME = 'Brut Bridge';
+	const URI = 'https://www.brut.media';
+	const DESCRIPTION = 'Returns 5 newest videos by category and edition';
+	const MAINTAINER = 'VerifiedJoseph';
+	const PARAMETERS = array(array(
+			'category' => array(
+				'name' => 'Category',
+				'type' => 'list',
+				'values' => array(
+					'News' => 'news',
+					'International' => 'international',
+					'Economy' => 'economy',
+					'Science and Technology' => 'science-and-technology',
+					'Entertainment' => 'entertainment',
+					'Sports' => 'sport',
+					'Nature' => 'nature',
+				),
+				'defaultValue' => 'news',
+			),
+			'edition' => array(
+				'name' => ' Edition',
+				'type' => 'list',
+					'values' => array(
+						'United States' => 'us',
+						'United Kingdom' => 'uk',
+						'France' => 'fr',
+						'India' => 'in',
+						'Mexico' => 'mx',
+				),
+				'defaultValue' => 'us',
+			)
+		)
+	);
+
+	const CACHE_TIMEOUT = 1800; // 30 mins
+
+	private $feedURI = '';
+
+	private $videoImageRegex = '/https:\/\/img\.brut\.media\/thumbnail\/(?:[a-z0-9-]+)-([a-z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+)-(square|landscape|portrait)(?:-auto)?\.jpg/';
+
+	public function collectData() {
+
+		$this->feedURI = self::URI . '/' . $this->getInput('edition') . '/' . $this->getInput('category');
+
+		$html = getSimpleHTMLDOM($this->feedURI)
+			or returnServerError('Could not request: ' . $this->feedURI);
+
+		$results = $html->find('div.results', 0);
+
+		foreach($results->find('li.col-6.col-sm-4.col-md-3.col-lg-2.px-2.pb-4') as $index => $li) {
+			$item = array();
+
+			if ($index > 5) {
+				break;
+			}
+
+			$videoPath = self::URI . $li->children(0)->href;
+
+			$videoPageHtml = getSimpleHTMLDOMCached($videoPath, 3600)
+				or returnServerError('Could not request: ' . $videoPath);
+
+			$videoImage = $videoPageHtml->find("meta[name=twitter:image]")[0]->content;
+			
+			preg_match($this->videoImageRegex, $videoImage, $matches)
```

Comment:
This operation fails if the file extension is `png` instead of `img`.
I strongly suggest parsing the image URL manually.
This works for me.

```php
/**
 * Extract video ID + type from twitter image
 *
 * Example (wrapped):
 *  https://img.brut.media/thumbnail/
 *  the-life-of-rita-moreno-2cce75b5-d448-44d2-a97c-ca50d6470dd4-square.jpg
 *  ?ts=1559337892
 */
$twimg = $videoPageHtml->find('meta[name="twitter:image"]', 0);
$fpath = parse_url($twimg->content, PHP_URL_PATH);
$fname = basename($fpath);
$fname = substr($fname, 0, strrpos($fname, '.'));
$parts = explode('-', $fname);

$videoId = implode('-', array_splice($parts, -6, 5));
$videoType = end($parts);
```

It doesn't rely on the URL, file extension or query parameters and therefore should work as long as the ID and video type are the last six parts of the file name.

Reply:
This is much better than my regex mess. I've had to make a small change so it works with URLs like `https://img.brut.media/thumbnail/restaurant-fights-food-waste-by-lowering-prices-f80d9f94-2c08-4187-a9ab-2788befeebbb-square-auto.jpg`. 

```
$twimg = $videoPageHtml->find('meta[name="twitter:image"]', 0);
$fpath = parse_url($twimg->content, PHP_URL_PATH);
$fname = basename($fpath);
$fname = substr($fname, 0, strrpos($fname, '.'));
$parts = explode('-', $fname);

if (end($parts) === 'auto') {
	$key = array_search('auto', $parts);
	unset($parts[$key]);
}

$videoId = implode('-', array_splice($parts, -6, 5));
$videoType = end($parts);
```

Thanks for the feedback.